### PR TITLE
feat: grapher page related research from DB content graph

### PR DESCRIPTION
**Pull request by mlbrgl**
This PR replaces the fortunejs content graph with a DB-based alternative, using the links stored in the `posts_links` and `posts_gdoc_links` tables.

This PR also continues the deprecation work started in #3166.

- [x] fully rebake site on staging
- [x] fixed malformed prominent link in content (ourworldindata.org/ instead of https://ourworldindata.org/)
	- [x] https://owid.cloud/wp/wp-admin/post.php?post=34103&action=edit
	- [x] https://owid.cloud/wp/wp-admin/post.php?post=34066&action=edit

**Inconsistency on how grapher chart links are stored in the DB**: there are a handful of chart links stored with the https://ourworldindata.org/grapher prefix. These disappear upon saving the containing articles, which indicates that they were created with an older version of the codebase. I'm then opting for not supporting them in the content graph.
```sql
    SELECT pgl.*, pg.published from posts_gdocs_links pgl
    JOIN posts_gdocs pg on pg.id = pgl.sourceId
    WHERE pgl.target LIKE "https://ourworldindata.org/grapher%"
```
 ### Testing links

Below are some testing links to grapher pages, backlinking to posts in different configurations.

- gdoc: https://ourworldindata.org/grapher/agricultural-export-subsidies
	- [x] http://localhost:3030/grapher/agricultural-export-subsidies
	- [x] http://staging-site-db-content-graph/grapher/agricultural-export-subsidies
- gdocs: https://ourworldindata.org/grapher/pollution-deaths-from-fossil-fuels
	- [x] http://localhost:3030/grapher/pollution-deaths-from-fossil-fuels
	- [x] http://staging-site-db-content-graph/grapher/pollution-deaths-from-fossil-fuels
- wp: https://ourworldindata.org/grapher/dalys-rate-from-all-causes
	- [x] http://localhost:3030/grapher/dalys-rate-from-all-causes
	- [x] http://staging-site-db-content-graph/grapher/dalys-rate-from-all-causes
- wp (with chart redirect): https://ourworldindata.org/grapher/age-standardized-death-rate-from-pm25-pollution-per-100000-vs-gdp-per-capita-int-
	- [x] http://localhost:3030/grapher/age-standardized-death-rate-from-pm25-pollution-per-100000-vs-gdp-per-capita-int-
	- [x] http://staging-site-db-content-graph/grapher/age-standardized-death-rate-from-pm25-pollution-per-100000-vs-gdp-per-capita-int-
- gdoc (with chart redirect): https://ourworldindata.org/grapher/population-long-run-with-projections?time=earliest..2100&country=~OWID_WRL
	- [x] http://localhost:3030/grapher/population-long-run-with-projections?time=earliest..2100&country=~OWID_WRL
	- [x] http://staging-site-db-content-graph/grapher/population-long-run-with-projections?time=earliest..2100&country=~OWID_WRL
- none: https://ourworldindata.org/grapher/death-rates-alcohol-drug-overdoses-by-age-who
	- [x] http://localhost:3030/grapher/death-rates-alcohol-drug-overdoses-by-age-who
	- [x] http://staging-site-db-content-graph/grapher/death-rates-alcohol-drug-overdoses-by-age-who

### CODE REVIEW

**Comment by mlbrgl** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3176?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #3165** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3165?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3166** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3166?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3176** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3176?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/3176?utm_source=stack-comment).

**Comment by mlbrgl** [comment]
Back to the drawing board, evaluating whether [`getReferencesByChartId`](https://github.com/owid/owid-grapher/blob/db-content-graph/adminSiteServer/apiRouter.ts#L195) could be a good fit.

**Comment by mlbrgl** [comment]
Turns out it is: #3179. Given the refactor in the next PR, this one then becomes more of a stepping stone than a final destination. Comments regarding the content graph backreferences query should rather be made in #3179.

File: db/model/Post.ts [hunk 1]
```diff
@@ -287,3 +291,223 @@ export const getBlockContentFromSnapshot = async (
...
+                    cr.chart_id = c.id
+            )
+        WHERE
+            c.id = ?
+            AND p.status = 'publish'
+            AND p.isGdocPublished = 0
+            AND p.type != 'wp_block'
+            -- note: we are not filtering by linkType to cast of wider net: if a post links to an
+            -- explorer having the same slug as the grapher chart, we want to surface it as
+            -- a "Related research" as it is most likely relevant.
```
Comment (danyx23) [inline db/model/Post.ts:327]: I don't quite get this comment - can you try to explain to me in different words what you mean?

**Review by danyx23 — APPROVED** [review]
Looks good! Two minor comments but neither is very important

File: db/model/Chart.ts [hunk 2]
```diff
@@ -350,3 +354,28 @@ export const getMostViewedGrapherIdsByChartType = async (
...
+
+    return db.queryMysql(`-- sql
+                SELECT
+                    charts.config->>"$.slug" AS slug,
+                    charts.config->>"$.title" AS title,
+                    charts.config->>"$.variantName" AS variantName,
+                    MAX(chart_tags.keyChartLevel) as keyChartLevel
+                FROM charts
+                INNER JOIN chart_tags ON charts.id=chart_tags.chartId
+                WHERE JSON_CONTAINS(config->'$.dimensions', '{"variableId":${variableId}}')
```
[the remark this answers is not in the document]
Comment (mlbrgl) [reply db/model/Chart.ts:375]: thanks, that makes sense. This was only [moved during a refactor](https://github.com/owid/owid-grapher/commit/c1f4dbdca11d496ad60b8a8ce7e256a0322fb472#diff-ab0604f7a26be8d1267ef4e5fcec2f3e4a321c61a83a763796ac0710b7cac81b) so I made an issue for it: https://github.com/owid/owid-grapher/issues/3203.

File: db/model/Post.ts (hunk 1 above)
Comment (mlbrgl) [reply db/model/Post.ts:327]: The idea here was to not restrict linkType to "grapher" only, so a post embedding an explorer sharing the same slug as the chart would also be returned (next to posts linking to the chart only). This was more of an edge case optimization which I ended up dropping when I merged references (in the grapher admin) with related posts (in the grapher page) in https://github.com/owid/owid-grapher/pull/3179/commits/a58fea7bb077183f97abbb0d2e97f4bbc337a733.

**Comment by mlbrgl** [comment]
### Merge activity

* **Feb 19, 12:02 PM EST**: @mlbrgl started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/3176).

**Pull request merged**